torchao/experimental/ops/mps/register.mm

```diff
@@ -17,22 +17,14 @@
 
 // LowBit Quantized Linear on MPS Backend
 template <int nbit>
-Tensor linear_mps_kernel(
+void check_linear_mps_args(
     const Tensor& A,
     const Tensor& B,
     int64_t group_size,
     const Tensor& SZ) {
-  auto M = A.size(0);
   auto N = B.size(0);
   auto K = A.size(1);
 
-  TORCH_CHECK(
-      A.is_mps(), __func__, "A is on ", A.device(), " but expected on mps");
-  TORCH_CHECK(
-      B.is_mps(), __func__, "B is on ", B.device(), " but expected on mps");
-  TORCH_CHECK(
-      SZ.is_mps(), __func__, "SZ is on ", SZ.device(), " but expected on mps");
-
   TORCH_CHECK(
```

Comment:
In a separate diff move to TORCHAO CHECK as scott suggested

Reply:
PR updated